core/src/main/kotlin/org/evomaster/core/solver/SMTLibZ3DbConstraintSolver.kt

```diff
@@ -56,13 +60,16 @@ class SMTLibZ3DbConstraintSolver(
      * @param sqlQuery The SQL query to solve.
      * @return A list of SQL actions that can be executed to satisfy the query.
      */
-    override fun solve(sqlQuery: String): List<SqlAction> {
-        val queryStatement = parseStatement(sqlQuery) // Parse SQL query
-        val smtLib = this.generator.generateSMT(queryStatement) // Generate SMTLib problem
-        val fileName = storeToTmpFile(smtLib) // Store SMTLib to a temporary file
-        val z3Response = executor.solveFromFile(fileName) // Solve using Z3
+    override fun solve(schemaDto: DbSchemaDto, sqlQuery: String, numberOfRows: Int): List<SqlAction> {
+        // TODO: Use memoized, if it's the same schema and query, return the same result and don't do any calculation
```

Comment:
@arcuri82  What do you think of this? Running the same query more than once extracts the same SMT solver, and as we are not using any seed the result is the same.

Reply:
hi @agusaldasoro  (and @jgaleotti ). if the query is the same, those could be cached, and executed only once. however, how often does that happen? I would assume most of the times the queries are parametric (based on inputs coming from the HTTP request). we could still use a cache, but need to make sure it is bound in size